### ISSUE TITLE
Improved ways to get raw output from the shell.

### ISSUE THREAD

**Issue by gnat** [Fizzadar/pyinfra#559]
## Is your feature request related to a problem? Please describe
So I'm often needing ways to branch python code based on actual raw output from a shell command (Use case: in-depth system audits).

`host.fact.command()` does this fairly well, but it lacks any ability to switch users or permissions on the fly. Operations have plenty of ways to do this using parameters such as `sudo=` `use_sudo_password=` `su_user=`... etc.

## Describe the solution you'd like
One or more of the following would be ideal.

* Give `host.fact.command()` some form of equivalent to the user switching parameters enjoyed by Operations.
* Allow us to get raw output from the `server.shell()` Operation.

**Comment by gnat** [comment]
For anyone following, there is a workaround using `su - user -c " "`. Perhaps not the most elegant solution but here's how it's done:

```python
from pyinfra import host
SUDO = True

output = host.fact.command("su - guy -c \"run command using guy user shell here\" ")
```

**Comment by Fizzadar** [comment]
I think this is  an absolute must-have; I'm going to implement this alongside the new [explicit fact imports](https://github.com/Fizzadar/pyinfra/pull/516) functionality, so can do:

```py
host.get_fact(AptSources, sudo=True)
```

Will have the neat side effect of allowing operations to customise fact global arguments they use (by default facts called by an operation inherit the operation args).

**Comment by Fizzadar** [comment]
Update on this: have implemented the change in https://github.com/Fizzadar/pyinfra/pull/516/commits/bc9d1182c8a76eb4171c4fb1ba5509e2ed2d1b66; need to add some tests and then should be good to go for `1.4` release shortly!

**Comment by Fizzadar** [comment]
Update on this: there's actually another way to do this (in 1.3) - calling `host.run_shell_command` directly should work here, with the added bonus that it won't run on every host, just the "current" one being executed (may or may not be beneficial). The [dynamic execution example](https://docs.pyinfra.com/en/1.x/examples/dynamic_execution_deploy.html) contains some example code for this.

**Comment by Fizzadar** [comment]
The new fact usage has now been released in `v1.4`!

**Comment by gnat** [comment]
Very cool @Fizzadar ! Thanks.
